### DISCUSSION TITLE
How to get data in the context menu?

### DISCUSSION THREAD

**Discussion by gabrielcarloto** [Flow-Launcher/Flow.Launcher#1304] [Q&A] [answered]
I found this (https://github.com/Flow-Launcher/Flow.Launcher/discussions/574#discussioncomment-1018794) but I don't know how to get this `ContextData`. I thought it would be something like this:

```js
const { method, parameters, ContextData } = JSON.parse(process.argv[2]);

if (method === "context_menu") {
  // do something with the context data
}
```

But the data sent from Flow in the context menu is

```json
{
  "method": "context_menu",
  "parameters": [
    null
  ],
  "settings": null
}
```

**Comment by taooceros** [comment]
If I remember correctly, the parameters will be something you sent with the results. There's a optional field of a result called `ContextData` that will be sent here for a selected result. So you need to include that for results to see the parameters here.

**Comment by Garulf** [comment, marked as answer]
Please see https://github.com/Flow-Launcher/docs/issues/49 for an example of Context Menu being used.

**Reply by gabrielcarloto** [reply]
It is still a bit confusing to me because I'm not familiar with python (my plugins are written in Typescript). Where does the list from `data` in `def context_menu(self, data)` come from? I know you passed the list in `'ContextData': ['context1', 'context2']`, but it is not sent in the JSON RPC. Like I said in my first question, the data sent from Flow is

```json
{
  "method": "context_menu",
  "parameters": [
    null
  ],
  "settings": null
}
```

(I got this by displaying the argv in the title of a result in the context menu: `Title: process.argv[2]`)
